### DISCUSSION TITLE
icmp PING module

### DISCUSSION THREAD

**Discussion by Andy1616** [prometheus/snmp_exporter#972] [Ideas]
hi all

slightly off topic - but I debugging SNMP settings to me it would be so helpful if I could have a icmp ping added to the SNMP requests...

Like an additional module, to me that would fit best.
Retry, timeout might be good to be configurable...

Any comments on that?
Arguments?

Or clearly keep on SNMP and icmp to be done in like blackbox only??

Happy to hear - thanks a lot

**Comment by SuperQ** [comment]
Prometheus provides an `up` metric, which should provide a better signal than ICMP.
